src/Compilers/CSharp/Portable/Compiler/ClsComplianceChecker.cs

```diff
@@ -158,22 +158,13 @@ public override void VisitNamespace(NamespaceSymbol symbol)
                 CheckMemberDistinctness(symbol);
             }
 
-            if (_compilation.Options.ConcurrentBuild)
```

Comment:
>if (_compilation.Options.ConcurrentBuild) [](start = 12, length = 41)

I am not comfortable with removing this parallelization completely. Even if we didn't add it based on specific scenario, similar approach was definitely proven to improve throughput for other similar cases (when we need to visit all namespaces and types recursively and do some analysis for them). Could we use some heuristic to disable parallelization only when it is likely cause a problem? For example, only when we have cancellable _cancellationToken and probably non-null _filterTree.